terraso_backend/tests/graphql/test_soil_id.py

```diff
@@ -0,0 +1,190 @@
+# Copyright © 2024 Technology Matters
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see https://www.gnu.org/licenses/.
+
+
+import pytest
+import structlog
+from graphene_django.utils.testing import graphql_query
+
+pytestmark = pytest.mark.django_db
+
+logger = structlog.get_logger(__name__)
+
+SOIL_MATCH_FRAGMENTS = """
+  fragment soilMatch on SoilMatchInfo {
+    score
+    rank
+  }
+  fragment soilInfo on SoilInfo {
+    soilSeries {
+      name
+      taxonomySubgroup
+      description
+      fullDescriptionUrl
+    }
+    ecologicalSite {
+      name
+      id
+      url
+    }
+    landCapabilityClass {
+      capabilityClass
+      subClass
+    }
+    soilData {
+      slope
+      depthDependentData {
+        depthInterval {
+          start
+          end
+        }
+        texture
+        rockFragmentVolume
+        colorHue
+        colorValue
+        colorChroma
+      }
+    }
+  }
+"""
+
+LOCATION_BASED_MATCHES_QUERY = (
+    """
+  query locationBasedSoilMatches($latitude: Float!, $longitude: Float!) {
+    soilId {
+      locationBasedSoilMatches(latitude: $latitude, longitude: $longitude) {
+        ...locationBasedSoilMatches
+      }
+    }
+  }
+  fragment locationBasedSoilMatches on LocationBasedSoilMatches {
+    matches {
+      dataSource
+      inMapUnit
+      match {
+        ...soilMatch
+      }
+      soilInfo {
+        ...soilInfo
+      }
+    }
+  }
+"""
+    + SOIL_MATCH_FRAGMENTS
+)
+
+
+def test_location_based_soil_matches_endpoint(client):
+    response = graphql_query(
+        LOCATION_BASED_MATCHES_QUERY, variables={"latitude": 0.0, "longitude": 0.0}, client=client
```

Comment:
Question: Would this fail once we have actual data coming in? I don't think there's soil at null island? :)

Reply:
i think you're right :thinking: https://www.openstreetmap.org/#map=10/0.0000/0.0000

& yes, this will get updated to real coordinates when there's real data!